### PR TITLE
feature/567_batches_within_transactions_ckan

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #567 
* 100% unit tests passed:
```
Tests run: 62, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
--------
row-mode
--------
```

test_row_ckan

```
-----------
column-mode
-----------
```

test_column_ckan

* Assignee @fgalan

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- [FEATURE] Add support to batch processing within a transaction in OrionMySQLSink (#567)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: ...in OrionMySQLSink

However, branch name mentions CKAN. Is this a typo?

File: doc/design/OrionCKANSink.md [hunk 2]
```diff
@@ -1,50 +1,63 @@
...
+    * `entityType`: Notified entity type.
+    * `attrName`: Notified attribute name.
+    * `attrType`: Notified attribute type.
+    * `attrValue`: In its simplest form, this value is just a string, but since Orion 0.11.0 it can be Json object or Json array.
+    * `attrMd`: It contains a string serialization of the metadata array for the attribute in Json (if the attribute hasn't metadata, an empty array `[]` is inserted).
+* `column`: A single row is upserted for all the notified context attributes. This kind of row will contain two fields per each entity's attribute (one for the value, called `<attrName>`, and other for the metadata, called `<attrName>_md`), plus four additional fields:
+    * `recvTime`: UTC timestamp in human-redable format ([ISO 8601](http://en.wikipedia.org/wiki/ISO_8601)).
+    * `fiwareservicePath`: The notified one or the default one.
+    * `entityId`: Notified entity identifier.
+    * `entityType`: Notified entity type.
```
Comment (fgalan) [inline doc/design/OrionCKANSink.md]: Any refactor in this part of the doc? Or there are relevant additions/deletions to delete? The diff is a bit noisy... :)

File: doc/design/OrionCKANSink.md [hunk 3]
```diff
@@ -237,25 +253,46 @@ A configuration example could be:
...
 
 [Top](#top)
 
-##<a name="section4"></a>Implementation details
-###<a name="section4.1"></a>`OrionCKANSink` class
+###<a name="section2.3"></a>Important notes
+####<a name="section2.3.1"></a>About the persistence mode
+Please observe not always the same number of attributes is notified; this depends on the subscription made to the NGSI-like sender. This is not a problem for the `row` persistence mode, since fixed 8-fields rows are upserted for each notified attribute. Nevertheless, the `column` mode may be affected by several rows of different lengths (in term of fields). Thus, the `column` mode is only recommended if your subscription is designed for always sending the same attributes, event if they were not updated since the last notification.
+
+In addition, when running in `column` mode, due to the number of notified attributes (and therefore the number of fields to be written within the Datastore) is unknown by Cygnus, the Datastore can not be automatically created, and must be provisioned previously to the Cygnus execution. That's not the case of the `row` mode since the number of fields to be written is always constant, independently of the number of notified attributes.
```
Comment (fgalan) [inline doc/design/OrionCKANSink.md]: can not -> cannot

(I'm not fully sure, but I think is the preferred form in English)

File: doc/design/OrionCKANSink.md (hunk 2 above)
Comment (frbattid) [inline doc/design/OrionCKANSink.md]: Just paragraphs reordering...

File: doc/design/OrionCKANSink.md (hunk 3 above)
Comment (frbattid) [inline doc/design/OrionCKANSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/7d6c2c61abac91e24c4abbe0b3abb559bf10e765

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/7d6c2c61abac91e24c4abbe0b3abb559bf10e765

File: doc/design/OrionCKANSink.md (hunk 2 above)
Comment (fgalan) [inline doc/design/OrionCKANSink.md]: NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**